akka-remote/src/main/scala/akka/remote/artery/ArterySubsystem.scala

```diff
@@ -67,30 +67,27 @@ private[remote] class ArterySubsystem(_system: ExtendedActorSystem, _provider: R
 
     val association =
       if (cached ne null) cached
-      else {
-        val association = getAssociation(remoteAddress)
-        association.associate()
-        recipient.cachedAssociation = association
-        association
-      }
+      else associate(remoteAddress)
 
     association.send(message, senderOption, recipient)
   }
 
-  private def getAssociation(remoteAddress: Address): Association = {
+  private def associate(remoteAddress: Address): Association = {
     val current = associations.get(remoteAddress)
     if (current ne null) current
     else {
-      val newAssociation = new Association(materializer, remoteAddress, transport)
-      val currentAssociation = associations.putIfAbsent(remoteAddress, newAssociation)
-      if (currentAssociation eq null) {
-        newAssociation
-      } else currentAssociation
+      associations.computeIfAbsent(remoteAddress, new java.util.function.Function[Address, Association] {
```

Comment:
Agreed. In my experience, if one does not clearly demarcate one's FIXMEs and do not scrap all the code at some point prior to merging, they tend to leak into the codebase—so faster and cheaper to annotate FIXMEs early and often :)